File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] ingredient) {
        
        int[] perfect = {1,2,3,1};

        int answer = 0;
        Stack<Integer> stack = new Stack<>(); //int형 스택 선언
        
        for(int i = 0; i<ingredient.length; i++){
            stack.push(ingredient[i]);
            
            if(stack.size() >= 4){
                if(stack.get(stack.size()-4) == 1 && stack.get(stack.size()-3) == 2 &&stack.get(stack.size()-2) == 3 && stack.get(stack.size()-1) == 1){
                    answer++;
                    
                    
                    stack.pop();stack.pop();stack.pop();stack.pop();
                }
            }
        }
        return answer;
    }
    
    
}

File: 프로그래머스/Lv.1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    
    let hamburger = [];
    
    let answer = 0;
    
    for(let a of ingredient){
        
        hamburger.push(a);
        
        if(hamburger.length >= 4){ //햄버거를 만들 수 있을만큼 재료가 쌓였을 때
            let len = hamburger.length
            if(hamburger[len-1] === 1 && hamburger[len-2] === 3 && hamburger[len-3] === 2 && hamburger[len-4] === 1){ //빵, 야채, 고기, 빵
                answer++;
                hamburger.pop();hamburger.pop();hamburger.pop();hamburger.pop(); //햄버거 만든 재료는 삭제
            }
        }
    }
    
    return answer;
}